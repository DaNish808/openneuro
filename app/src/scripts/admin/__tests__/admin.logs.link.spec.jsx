import LogLink from '../admin.logs.link'

describe('admin/LogLink', () => {
  it('renders successfully', () => {
    const log = { data: {} }
    const wrapper = shallow(<LogLink log={log} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('links to jobs', () => {
    const log = {
      data: {
        job: {
          datasetId: 'someid',
          snapshotId: 'anotherid',
          appLabel: 'someapp',
          appVersion: '1',
          datasetLabel: 'somedataset',
        },
      },
    }
    const wrapper = shallow(<LogLink log={log} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('Link').length).toBe(1)
  })
  it('links to datasets', () => {
    const log = {
      data: {
        dataset: {
          datasetId: 'someid',
          datasetLabel: 'somedataset',
        },
      },
    }
    const wrapper = shallow(<LogLink log={log} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('Link').length).toBe(1)
  })
})
