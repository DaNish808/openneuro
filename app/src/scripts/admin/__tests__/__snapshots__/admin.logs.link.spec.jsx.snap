// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin/LogLink links to datasets 1`] = `
<Link
  activeClassName="active"
  className=""
  params={
    Object {
      "datasetId": "someid",
    }
  }
  to="dataset"
>
  somedataset
</Link>
`;

exports[`admin/LogLink links to jobs 1`] = `
<Link
  activeClassName="active"
  className=""
  params={
    Object {
      "datasetId": "someid",
      "snapshotId": "anotherid",
    }
  }
  query={
    Object {
      "app": "someapp",
      "job": undefined,
      "version": "1",
    }
  }
  to="snapshot"
>
  somedataset
   
  <br />
   
  someapp
  :
  1
</Link>
`;

exports[`admin/LogLink renders successfully 1`] = `<span />`;
